utils/wasm-gen/src/utils.rs

```diff
@@ -464,17 +464,63 @@ pub fn inject_critical_gas_limit(module: Module, critical_gas_limit: u64) -> Mod
     module
 }
 
-pub(crate) fn hash_with_value_to_vec(hash_with_value: &HashWithValue) -> Vec<u8> {
-    let address_data_size = mem::size_of::<HashWithValue>();
-    let address_data_slice = unsafe {
-        // # Safety:
-        // The `unsafe` block constructs raw bytes vector of an existing rust struct
-        // received by reference.
-        slice::from_raw_parts(
-            hash_with_value as *const HashWithValue as *const u8,
-            address_data_size,
-        )
-    };
+pub(crate) struct WasmWords(Vec<i32>);
+
+impl WasmWords {
+    pub(crate) fn new(data: impl AsRef<[u8]>) -> Self {
+        let data = data.as_ref();
+
+        let data_size = data.len();
+        let wasm_ptr_size = mem::size_of::<i32>();
+
+        if data_size % wasm_ptr_size != 0 {
+            panic!("data size isn't multiply of wasm word size")
+        }
+
+        let words = data
+            .chunks(wasm_ptr_size)
```

Comment:
since it was checked before
```suggestion
            .chunks_exact(wasm_ptr_size)
```

Reply:
done [2668531](https://github.com/gear-tech/gear/pull/3625/commits/266853145bfc59843fb3ad8b9407d52e1a9559b4)